src/transformer/mockDefiner/mockDefiner.ts

```diff
@@ -213,8 +232,9 @@ export class MockDefiner {
   private _getMockFactoryIdForIntersections(declarations: ts.Declaration[], intersectionTypeNode: ts.IntersectionTypeNode): string {
     const thisFileName: string = this._fileName;
 
-    if (this._factoryIntersectionCache.has(declarations)) {
-      return this._factoryIntersectionCache.get(declarations);
+    const factoryIntersection: string | undefined = this._factoryIntersectionCache.get(declarations);
```

Comment:
I preferred the old version. Does it complain typescript? Can we cast it considering that if has works the get should work?

Reply:
I changed it because `has` doesn't act like a type guard and the `get` would result in `string | undefined` in either case. One thing I missed though, is that the change doesn't consider empty strings - however, I don't think that is a valid value anyway. `has` does check that implicitly.

Reply:
Perhaps, we should allow the ESLint `@typecript-eslint/no-non-null-assertion`-rule. That would allow us to do: `this._factoryIntersectionCache.get(declarations)!`.